### PR TITLE
Switch to storage gateway from minio and CustomEnvs

### PR DESCRIPTION
**Pull request by oldgiova**
* Replaced minio specific gateway with a generic storage_gateway service, that could work for both minio and Amazon S3
* Added customEnvs options

Ticket: MC-6856

### CODE REVIEW

File: mender/templates/api-gateway/deployment.yaml [hunk 1]
```diff
@@ -94,6 +94,12 @@ spec:
         resources:
 {{ toYaml .Values.api_gateway.resources | indent 10 }}
 
+        env:
+{{- if .Values.api_gateway.storage_proxy.enabled }}
+          - name: MENDER_STORAGE_PROXY
+            value: {{ include "mender.storageProxyUrl" . }}
+{{- end }}
```
Comment (alfrunes) [inline mender/templates/api-gateway/deployment.yaml:101]: What is this environment variable for?

File: mender/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## master
 * Fix mongodb uri creation when using the mongodb subchart and replicast architecture
+* Replaced minio specific gateway with a generic `storage_gateway` service, that could
```
Comment (alfrunes) [inline mender/CHANGELOG.md:5]: This is a breaking change -- should deserve a major version bump in the `Chart.yaml`

File: mender/values.yaml [hunk 3]
```diff
@@ -69,6 +69,11 @@ default:
     enabled: false
     minAvailable: 1
 
+  # custom default envs for every service
+  customEnvs: []
```
Comment (alfrunes) [inline mender/values.yaml:73]: Can we `s/customEnvs/env/` to align it with Alvaldi?

File: mender/CHANGELOG.md (hunk 2 above)
Comment (oldgiova) [reply mender/CHANGELOG.md:5]: Right! Why not to use a drop-in replacement?

Keep using both the keys:
```
api_gateway:
  minio:
    enabled: true
```
and:
```
api_gateway:
  storage_proxy:
    enabled: false
```

File: mender/values.yaml (hunk 3 above)
Comment (oldgiova) [reply mender/values.yaml:73]: I tried, but I found two issues:
* The Alvaldi [utility](https://github.com/NorthernTechHQ/alvaldi-helm/blob/main/alvaldi/templates/_helpers.tpl#L261) is a bit different: it's not merging the envs (afaik) and I need a merge
* many keys `<service>.env` are already in place, with different notations:
  ```
    env:
      TENANTADM_MIDDLEWARE: prod
      TENANTADM_SERVER_PRIV_KEY_PATH: /etc/tenantadm/rsa/private.pem
  ```
But the Alvaldi notation implies:
```
env:
  - name: TENANTADM_MIDDLEWARE
    value: prod
  - name: TENANTADM_SERVER_PRIV_KEY_PATH
    value: /etc/tenantadm/rsa/private.pem
```

and this requires a delicate migration and a breaking change. Imho we should keep it simple with a key different than `env`, but let me know what you prefer.
If you want to align with Alvaldi, I'd prefer moving Alvaldi to `s/env/customEnvs/` or similar

Comment (alfrunes) [reply mender/values.yaml:73]: Ahh, ok. I was not aware.

File: mender/templates/api-gateway/deployment.yaml (hunk 1 above)
Comment (alfrunes) [reply mender/templates/api-gateway/deployment.yaml:101]: Indeed there it's used in the Traefik [dynamic configuration[(https://github.com/mendersoftware/saas/blob/master/kubernetes/mender_stack/gateway/config-mender-api-gateway-traefik.yaml#L7), but this environment variable does not affect the config here. With helm we can template this extra router directly although I would prefer having this at the ingress level and not inside Traefik.
IOW, I think we can drop this one and rely on the ingress.

Comment (oldgiova) [reply mender/templates/api-gateway/deployment.yaml]: Dropped, thanks

File: mender/CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
+* Added generic `storage_proxy` service, that could
+  work for both minio and Amazon S3, and it's going to replace the `api_gateway.minio` configuration.
+* **Deprecations**:
+  * `api_gateway.minio` is deprecated in favor of `api_gateway.storage_proxy`.
+    This entry could be used, but it is no longer maintained, and could be removed
+    in future releases.  
+    **How to upgrade**: 
+    * set `api_gateway.minio.enabled=false`
+    * set `api_gateway.storage_proxy.enabled=true`
+    * set `api_gateway.storage_proxy.url` to the external storage url that you want to map externally. For example `https://fleetstorage.example.com`.
```
Comment (oldgiova) [inline mender/CHANGELOG.md:15]: I'm not 100% sure about this statement

File: mender/CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,19 @@
 # Mender Helm chart
 
-## master
+## 5.5.0
 * Fix mongodb uri creation when using the mongodb subchart and replicast architecture
+* Added customEnv option to set default or per-service custom env variables
+* Added generic `storage_proxy` service, that could
+  work for both minio and Amazon S3, and it's going to replace the `api_gateway.minio` configuration.
+* **Deprecations**:
```
Comment (oldgiova) [inline mender/CHANGELOG.md:8]: I chose a soft deprecation: `api_gateway.minio` is still enabled (and default choice) but it's considered deprecated.

**Review by alfrunes — APPROVED** [review]

**Pull request merged**